startup/GafferScene/renderPassTypes.py

```diff
@@ -0,0 +1,520 @@
+	processor["__cyclesCatcherAttributes"]["filter"].setInput( processor["__catcherAndCasterFilter"]["catchers"] )
+
+	# On shadow casters we override `cycles:is_shadow_catcher` off to ensure
+	# casters that are descendants of catchers do not inherit the attribute.
+	processor["__cyclesCasterAttributes"] = GafferScene.CustomAttributes()
+	processor["__cyclesCasterAttributes"]["in"].setInput( processor["__cyclesCatcherAttributes"]["out"] )
+	processor["__cyclesCasterAttributes"]["attributes"].addChild( Gaffer.NameValuePlug( "cycles:is_shadow_catcher", Gaffer.BoolPlug( defaultValue = False ) ) )
+	processor["__cyclesCasterAttributes"]["filter"].setInput( processor["__catcherAndCasterFilter"]["casters"] )
+
+	processor["__lightsSet"] = GafferScene.SetFilter()
+	processor["__lightsSet"]["setExpression"].setValue( "__lights" )
+
+	# Disabling shadow visibility on lights in 3Delight cause them to not cast shadows, so we must override
+	# the `__allShadowExcluded` global shadow visibility attribute if it is enabled.
+	processor["__lightShadowVisibility"] = GafferScene.AttributeTweaks()
+	processor["__lightShadowVisibility"]["in"].setInput( processor["__copyAttributes"]["out"] )
+	processor["__lightShadowVisibility"]["enabled"].setInput( processor["__allShadowExcluded"]["enabled"] )
+	processor["__lightShadowVisibility"]["tweaks"].addChild( Gaffer.TweakPlug( "", Gaffer.BoolPlug( defaultValue = True ), mode = Gaffer.TweakPlug.Mode.CreateIfMissing ) )
+	processor["__lightShadowVisibility"]["tweaks"][0]["name"].setValue( "dl:visibility.shadow" )
+	processor["__lightShadowVisibility"]["filter"].setInput( processor["__lightsSet"]["out"] )
+
+	processor["__rendererSwitch"] = Gaffer.NameSwitch()
+	processor["__rendererSwitch"].setup( GafferScene.ScenePlug() )
+	processor["__rendererSwitch"]["selector"].setInput( processor["renderer"] )
+	processor["__rendererSwitch"]["in"].resize( 3 )
+	processor["__rendererSwitch"]["in"]["in0"]["value"].setInput( processor["__copyAttributes"]["out"] )
+
+	processor["__rendererSwitch"]["in"]["in1"]["name"].setValue( "Cycles" )
+	processor["__rendererSwitch"]["in"]["in1"]["value"].setInput( processor["__cyclesCasterAttributes"]["out"] )
+
+	processor["__rendererSwitch"]["in"]["in2"]["name"].setValue( "3Delight*" )
+	processor["__rendererSwitch"]["in"]["in2"]["value"].setInput( processor["__lightShadowVisibility"]["out"] )
+
+	# Redirect the Cycles "shadow_catcher" AOV to beauty to more closely match
+	# Arnold and 3Delight behaviour.
+	processor["__redirectOutputExpression"] = Gaffer.Expression()
+	processor["__redirectOutputExpression"].setExpression(
+		inspect.cleandoc(
+			"""
+			options = parent["__cyclesCasterAttributes"]["out"]["globals"]
+			for o in options.keys() :
+				if o.startswith( "output:" ) :
+					if options[o].getType() != "ieDisplay" and options[o].getData() == "rgba" :
+						options[o].setData( "shadow_catcher" )
+
+			parent["__rendererSwitch"]["in"]["in1"]["value"]["globals"] = options
+			"""
+		)
+	)
```

Comment:
Could this be split into a separate adaptor, demonstrating that additional renderer-specific stuff can be done non-intrusively on top of the core adaptors?

Reply:
I've split the Cycles output remapping into a separate shadow pass processor in 804e83e872638c98bf98ce65b93950bffc37c713. Or were you thinking of actually registering it as a separate adaptor outside of the RenderPassTypeAdaptor?